### PR TITLE
shading dependencies in pulsar client

### PR DESCRIPTION
**Pull request by jerrypeng**
Minimizing the number of dependencies in Pulsar client by shading them

### CODE REVIEW

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

File: pulsar-client-shaded/pom.xml [hunk 1]
```diff
@@ -217,6 +220,10 @@
                   <pattern>org.objenesis</pattern>
                   <shadedPattern>org.apache.pulsar.shade.org.objenesis</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>org.yaml</pattern>
+                  <shadedPattern>org.apache.pulsar.shade.org.yaml</shadedPattern>
+                </relocation>
```
Comment (merlimat) [inline pulsar-client-shaded/pom.xml]: @jerrypeng We also need the same in `pulsar-client-admin-shaded` and `pulsar-client-kafka`

Comment (jerrypeng) [reply pulsar-client-shaded/pom.xml]: I have shaded the dependencies in those modules as well

**Pull request merged**